src/unix.rs

```diff
@@ -385,8 +397,26 @@ impl Helper {
     }
 }
 
-fn is_valid_fd(fd: c_int) -> bool {
-    unsafe { libc::fcntl(fd, libc::F_GETFD) != -1 }
+fn check_fd(fd: c_int) -> bool {
+    #[cfg(feature = "check_pipe")]
```

Comment:
```suggestion
    #[cfg(feature = "check_pipe")]
```

IMHO this check should not be optional

Reply:
It was removed with one reason. And there isn't such check in gnu make.

I'm not sure about default behavior though.

Reply:
[POSIX jobserver](https://www.gnu.org/software/make/manual/html_node/POSIX-Jobserver.html) requires the jobserver on Unix to be implemented with either named pipe or fifo, so if they pass something else, it's incorrect and should be reject, at least that's how I interpret the spec.